File: MANGAKYO/samehadaku.py
from flask import Flask, render_template, request, jsonify

app = Flask(__name__)

# Daftar komik yang akan ditampilkan
comics = [
    {"title": "Blue Lock", "chapter": "Chapter 281"},
    {"title": "DANDADAN", "chapter": "Chapter 173"},
    {"title": "One Piece", "chapter": "Chapter 1131"},
    {"title": "Solo Leveling: Ragnarok", "chapter": "Chapter 27"},
]

@app.route('/')
def home():
    return render_template('index.html')

@app.route('/search', methods=['POST'])
def search():
    search_text = request.json.get('searchText', '').lower()
    filtered_comics = [comic for comic in comics if search_text in comic['title'].lower()]
    return jsonify(filtered_comics)

if __name__ == '__main__':
    app.run(debug=True)

File: MANGAKYO/samehadaku.java
import org.springframework.web.bind.annotation.*;
import java.util.ArrayList;
import java.util.List;

@RestController
public class ComicController {

    private List<Comic> comics;

    public ComicController() {
        comics = new ArrayList<>();
        comics.add(new Comic("Blue Lock", "Chapter 281"));
        comics.add(new Comic("DANDADAN", "Chapter 173"));
        comics.add(new Comic("One Piece", "Chapter 1131"));
        comics.add(new Comic("Solo Leveling: Ragnarok", "Chapter 27"));
    }

    @PostMapping("/search")
    public List<Comic> searchComics(@RequestBody SearchRequest request) {
        List<Comic> result = new ArrayList<>();
        for (Comic comic : comics) {
            if (comic.getTitle().toLowerCase().contains(request.getSearchText().toLowerCase())) {
                result.add(comic);
            }
        }
        return result;
    }
}

class Comic {
    private String title;
    private String chapter;

    public Comic(String title, String chapter) {
        this.title = title;
        this.chapter = chapter;
    }

    public String getTitle() {
        return title;
    }

    public String getChapter() {
        return chapter;
    }
}

class SearchRequest {
    private String searchText;

    public String getSearchText() {
        return searchText;
    }

    public void setSearchText(String searchText) {
        this.searchText = searchText;
    }
}
